packages/elements/src/list/elements/list.ts

```diff
@@ -0,0 +1,687 @@
+    const backupElement = this.tabbableItems[0];
+    return nextElement ? this.itemFromElement(nextElement) : backupElement ? this.itemFromElement(backupElement) : undefined;
+  }
+
+  /**
+   * Clears any highlighted item
+   * @returns {void}
+   */
+  protected clearHighlighted (): void {
+    this.queryItemsByPropertyValue('highlighted', true)
+      .forEach(item => this.composer.setItemPropertyValue(item, 'highlighted', false));
+  }
+
+  /**
+   * Highlights a single item.
+   * Used for navigation.
+   * @param item Item to highlight
+   * @param scrollToItem Scroll the item into view?
+   * @returns {void}
+   */
+  protected highlightItem (item?: T, scrollToItem = false): void {
+    if (item) {
+      this.clearHighlighted();
+      this.composer.setItemPropertyValue(item, 'highlighted', true);
+
+      if (this.tabIndex >= 0) {
+        const id = getItemId(this.renderer.key, item.value);
+        this.setAttribute('aria-activedescendant', id);
+      }
+
+      scrollToItem && this.scrollToItem(item);
+    }
+  }
+
+  /**
+   * Gets the available tabbable elements
+   */
+  protected get tabbableItems (): HTMLElement[] {
+    return Array.from(this.children).filter((el): el is HTMLElement => {
+      if (el instanceof HTMLElement) {
+        const role = el.getAttribute('role');
+        const isEnabled = !el.hasAttribute('disabled');
+        const isOption = role ? ['option', 'treeitem'].includes(role) : false;
+
+        return isOption && isEnabled;
+      }
+      return false;
+    });
+  }
+
+  /**
+   * Returns the current focused element
+   */
+  protected get activeElement (): HTMLElement | null {
+    const el = (this.getRootNode() as ShadowRoot | HTMLDocument).activeElement as HTMLElement | null;
+    const itemEl = this.findItemElementFromTarget(el);
+    if (itemEl && this.tabbableItems.includes(itemEl)) {
+      return itemEl;
+    }
+    return null;
+  }
+
+  /**
+   * Returns the current focused element
+   */
+  protected get highlightElement (): HTMLElement | null {
+    const item = this.queryItemsByPropertyValue('highlighted', true)[0];
+    return item ? this.elementFromItem(item) || null : null;
+  }
+
+  /**
+   * Tries to select the current highlighted element
+   * @returns {void}
+   */
+  protected triggerActiveItem (): void {
+    const el = this.activeElement || this.highlightElement;
+    const item = el && this.itemFromElement(el);
+    item && this.selectItem(item) && this.fireSelectionUpdate();
+  }
+
+  /**
+   * Scroll to list item element
+   * @param item Data item to scroll to
+   * @returns {void}
+   */
+  public scrollToItem (item: T): void {
+    const element = this.elementFromItem(item);
+    if (element) {
+      const min = this.scrollTop;
+      const max = this.scrollTop + this.clientHeight - element.offsetHeight;
+      const pos = element.offsetTop;
+      pos > max ? this.scrollTop = element.offsetTop - this.clientHeight + element.offsetHeight
```

Comment:
We need to set a variable here rather than updating the scrollTop value in the switch. Then use the variable to update the scrollTop position. This should fix the sonarcloud issue.

Reply:
```suggestion
      const scrollTop = pos > max ? element.offsetTop - this.clientHeight + element.offsetHeight : pos < min ? element.offsetTop : 0;
      this.scrollTop = scrollTop;
```

Something like this, or better, if you can clean up the calulations. Just needs a tidy up :)

Reply:
Updated. I just found out that this method never really work as a standalone public method. It uses an object as a key in `elementMap` when passing a new item object to `scrollToItem`, it would be a different reference value.